rust/zypp-agama/src/lib.rs

```diff
@@ -64,36 +65,90 @@ where
     Some(progress_callback::<F>)
 }
 
-pub fn init_target<F>(root: &str, progress: F) -> ZyppResult<()>
+// Mutex for the Zypp C API which is not thread safe.
+// TODO: guard some (global) context instead of ().
+//static ZYPP_MUTEX: Mutex<*mut zypp_agama_sys::Zypp> = Mutex::new(std::ptr::null_mut());
```

Comment:
So, if I try to put the zypp pointer into the Mutex, the compiler complains that a pointer cannot be sent safely across threads.
Is it good enough with `Mutex<()>` or am I overlooking a race?

Reply:
maybe you should instead of plain pointer use Rc? https://doc.rust-lang.org/std/rc/struct.Rc.html

Reply:
`Rc` is still single threaded, maybe `Arc` would work

Reply:
Question is if we need that multiple threads? If it cannot be send to another thread it can mean that we want to have zypp in single thread and react to messages ( which is what @imobachgs doing ).